File: flutter/lib/domain/use_case/deck/get_flashcards_from_deck_use_case.dart
import 'package:flashcard/domain/repository/flashcard_repository.dart';

import '../../entities/deck.dart';
import '../../entities/flashcard.dart';

class GetFlashcardsFromDeckUseCase {
  final FlashcardRepository _flashcardRepository;

  GetFlashcardsFromDeckUseCase(this._flashcardRepository);

  Future<List<Flashcard>> call(Deck deck) async {
    if (deck.id == null) {
      throw Exception("Deck ID is null. Cannot retrieve flashcards.");
    }
    return await _flashcardRepository.getFlashcardsByDeckId(deck.id!);
  }
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/usecase/GetFlashcardsFromDeckUseCase.kt
package com.synergyboat.flashcardAi.domain.usecase

import com.synergyboat.flashcardAi.domain.entities.Flashcard
import com.synergyboat.flashcardAi.domain.repository.FlashcardRepository

class GetFlashcardsFromDeckUseCase(
    private val repository: FlashcardRepository
) {
    suspend operator fun invoke(deckId: Long): List<Flashcard> {
        if (deckId <= 0) {
            throw IllegalArgumentException("Deck ID must be greater than zero")
        }
        return repository.getFlashcardsByDeck(deckId)
    }
}